eth/watchers/timewatcher.go

```diff
@@ -122,6 +122,12 @@ func (tw *TimeWatcher) Watch() error {
 		return fmt.Errorf("error fetching initial transcoderPoolSize err=%v", err)
 	}
 
+	lastSeenBlock, err := tw.watcher.GetLatestBlock()
+	if err != nil {
+		return fmt.Errorf("error fetching last seen block err=%v", err)
+	}
+	tw.setLastSeenBlock(lastSeenBlock.Number)
```

Comment:
I think `lastSeenBlock` can be nil if the node has a fresh DB. We should add a nil check and if it is nil we can use a sane default value like `big.NewInt(0)` which should only be temporary until backfilling updates the value to the latest block number.

Reply:
There's no need to set initialise this value I think as we have a nil check in the handler.

```
func (tw *TimeWatcher) handleBlockNum(event *blockwatch.Event) {
	last := tw.LastSeenBlock()
	new := event.BlockHeader.Number
	if last == nil || last.Cmp(new) != 0 {
		tw.setLastSeenBlock(new)
		tw.blockSubFeed.Send(new)
	}
}
```

Reply:
`handleBlockNum()` is only called when a `blockwatch.Event` is received via the block watcher subscription. This piece of code is run before the block watcher subscription is created and the value returned by `tw.watcher.GetLatestBlock()` could be nil (i.e. if the node has a fresh DB).

Reply:
Yes that's true but that doesn't really matter there's three scenarios that are all covered as is.

- Node has a fresh DB
  1. initialise `tw.lastSeenBlock = nil`
  2. backfill all blocks from the current round
  3. `tw.lastSeenBlock` is now set to the last seen backfilled block

- Node has a DB with an outdated `lastSeenblock`
 1. Initialise `tw.lastSeenBlock` to the last seen block in the DB
 2. If the block is in the current round start backfilling else backfill from the current round start block
 3. Set `tw.lastSeenblock` to the last backfilled block

- Node has a DB with a last seen block >= the actual last seen block
 1. Initialise `tw.LastSeenBlock` to the latest seen block
 2. Don't backfill
 
None of these cases will end up with a `tw.lastSeenBlock = nil` by the time node startup is complete.

Reply:
If the node has a fresh DB, `tw.watcher.GetLatestBlock()` the `lastSeenBlock` variable will be set to nil. Then `lastSeenBlock.Number` will trigger a nil pointer error. All this happens before backfilling happens.

Reply:
Ah right ! I missed that part 😅

Reply:
fixup cc20510 & unit test af853a1